### PR TITLE
76 set legend

### PR DESCRIPTION
**Pull request by pchelle**
I was hoping to finish the modifications of this pull request before you merge the previous one

### CODE REVIEW

**Comment by msevestre** [comment]
Sorry . Please add WIP (work in progress) or something like that in the
title.
Also please specify that it's a WIP in the PR description. No way to see
otherwise;-)

File: R/datamapping-groupings.R [hunk 1]
```diff
@@ -14,16 +14,17 @@ Groupings <- R6::R6Class(
                               fill = NULL,
                               linetype = NULL,
                               shape = NULL) {
-      if (!"GroupMapping" %in% class(color)) {
+
+      if (isOfType(color, "GroupMapping")) {
         color <- GroupMapping$new(group = color)
       }
-      if (!"GroupMapping" %in% class(fill)) {
+      if (isOfType(fill, "GroupMapping")) {
```
Comment (msevestre) [inline R/datamapping-groupings.R]: FYI. You can also use isOfType(fill, GroupMapping) (Without "" using type directly)
Also the code is wrong. Is should be `!isOfType(xxx)`

File: R/plotconfiguration-pkratio.R [hunk 2]
```diff
@@ -64,17 +51,21 @@ PKRatioPlotConfiguration <- R6::R6Class(
...
-          size = 1, # To be updated with current Theme
-          show.legend = TRUE
-        )
-      
+        mapping = aes(
+          x = mapData$x, y = mapData$y,
+          color = mapData[, color],
+          shape = mapData[, shape]
+        ),
+        size = 1, # To be updated with current Theme
```
Comment (msevestre) [inline R/plotconfiguration-pkratio.R]: add TODO everytime something needs to be done so that we can find it again

File: R/themes.R [hunk 3]
```diff
@@ -41,14 +42,15 @@ themeAesProperties$tlf <- list(
   shape = c("circle", "square", "diamond", "triangle"),
   linetype = c("solid", "dashed", "dotted")
 )
+themeAesProperties$tlf$fill <- themeAesProperties$tlf$color
 
 themeAesProperties$bw <- list(
   color = c("#000000", "#999999", "#555555"),
   size = seq(1, 10),
   shape = c("circle", "square", "diamond", "triangle"),
```
Comment (msevestre) [inline R/themes.R]: probably would make sense to define those guys as enum to

**Comment by pchelle** [comment]
Yes. You can merge the current pull request.

**Pull request merged**